MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/BuyClient.java

```diff
@@ -231,6 +238,46 @@ public void setWebReturnToLabel(String webReturnToLabel) {
     }
 
     /**
+     * Enables Android Pay support in the {@link BuyClient}
+     * @param androidPayPublicKey The base64 encoded public key associated with Android Pay.
+     */
+    public void enableAndroidPay(String androidPayPublicKey) {
+        if (TextUtils.isEmpty(androidPayPublicKey)) {
+            throw new IllegalArgumentException("androidPayPublicKey cannot be empty");
+        }
+
+        byte[] digest;
+        try {
+            MessageDigest messageDigest = MessageDigest.getInstance("SHA-256");
+            digest = messageDigest.digest(androidPayPublicKey.getBytes("UTF-8"));
+        } catch (NoSuchAlgorithmException | UnsupportedEncodingException e) {
+            // Do not enable Android Pay if the hash could not be computed
+            return;
+        }
+
+        // Enable Android Pay by setting the hash and key
+        this.androidPayPublicKeyHash = Base64.encodeToString(digest, Base64.DEFAULT);
+        this.androidPayPublicKey = androidPayPublicKey;
+    }
+
+    public boolean androidPayIsEnabled() {
+        return !TextUtils.isEmpty(androidPayPublicKey);
+    }
+
+    public void disableAndroidPay() {
```

Comment:
Also didn't find any logic that will disable android pay in case if server returns some error like key is not valid etc.